examples/api/lib/material/context_menu/editable_text_toolbar_builder.0.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This example demonstrates showing the default buttons, but customizing their
+// appearance.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  MyApp({super.key});
+
+  final TextEditingController _controller = TextEditingController(
+    text: 'Right click or long press to see the menu with custom button.',
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Custom button appearance'),
+        ),
+        body: Center(
+          child: Column(
+            children: <Widget>[
+              const SizedBox(height: 20.0),
+              TextField(
+                controller: _controller,
+                contextMenuBuilder: (BuildContext context, EditableTextState editableTextState, Offset primaryAnchor, [Offset? secondaryAnchor]) {
```

Comment:
It could be tempting for people to try to directly/indirectly use `editableTextState.context` and trigger asserts.

Reply:
True... Is there anything I can do about it?  It would be way overkill to try to create a mixin that doesn't include the BuildContext or something.

Reply:
Thanks for this comment, I spent a bunch of time refactoring to decouple from EditableText.  I definitely want this context menu system to be easily usable by people that are building custom editors etc. and not using EditableText.

  * I moved from builder widgets to functions where possible.
    - getEditableButtonItems, etc. instead of a StatelessWidget with a builder.
  * getEditableButtonItems just takes the required parameters directly, not EditableTextState.
  * getEditableTextButtonItems exists as a convenience and it calls out to getEditableButtonItems.
  * canCut etc. are back to being the responsibility of EditableText.
  * This all applies to SelectableRegion as well (getSelectableButtonItems doesn't depend on SelectableRegionState).

I didn't use TextSelectionDelegate because it seemed to be bigger than what I needed.  I think it will be easier for people not using EditableText/SelectableRegion to just pass in the parameters directly.

Reply:
@chunhtai I think you're right and that's better.  I've updated the PR to pass button items instead of states.

Originally I didn't do this because I didn't want to generate the button items if users didn't want to use them, but I think the work required is very minimal, and users likely do want to use them anyway.

This also means that SelectableRegionState can remain private.  I'll keep an eye out in case people want to access some of that state when they generate their context menus but I think the selection will be enough.

Reply:
I've moved back to passing EditableTextState, details in this comment: https://github.com/flutter/flutter/pull/107193#issuecomment-1268749813